website/docs/configuration.md

```diff
@@ -3,8 +3,77 @@ id: configuration
 title: Configuration
 ---
 
-TODO: Cover the mandatory configuration options here.
+Docusaurus asks for only a few fields to configure your site.
 
-#### References
+## `docusaurus.config.js`
 
-- https://v1.vuepress.vuejs.org/guide/basic-config.html
+When you create your site using `docusaurus init`, the `docusaurus.config.js` file is initialized alongside the template.
+
+### Configurable fields
+
+`docusaurus.config.js` contains configurable fields for:
+
+- Meta data of your site
+  - `title: string`
+  - `tagline: string`
+  - `favicon: string` relative url to your site's logo
+- Deployment fields for GitHub Pages, see [deployment configurations](/docs/deployment#docusaurusconfigjs-settings)
+  <!-- TODO: explain that theme configurations will be consumed by the theme, and link to theme doc -->
+- Theme configurations, see [theme configurations](#)
+  <!-- TODO: explain that preset configurations will be used to define presets of the site, and link to doc -->
+- Preset configurations, see [preset configurations](#)
+
+```js
+module.exports = {
+  // Site meta
+  title: 'Docusaurus',
+  tagline: '⚡️ Painless static site generator',
+  favicon: 'img/docusaurus.ico',
+
+  // Deployment fields
+  organizationName: 'facebook',
+  projectName: 'docusaurus',
+  baseUrl: '/',
+  url: 'https://v2.docusaurus.io',
+
+  themeConfig: {
+    // theme specific configurations
+  },
+  presets: [
+    // presets configurations
+  ],
+};
+```
+
+**Note**: `docusaurus.config.js` is not to be extended.
```

Comment:
When i read this, I feel confused. I think user doesnt need to know much about this. Actually we also allow customField in config. If they want to add new field, they should add it to customfield first